### PR TITLE
use only one makefile to build all components

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind feature

**What this PR does / why we need it**:
now every component has own makefile. Sometimes it is difficult to share some common operations。
This pr will use one makefile to build all components

To get instructions for each target, add the HELP = y parameter

for all target: 
```
make all HELP=y
```

for verify target:
```
make verify HELP=y
```

for test target:
```
make test HELP=y
```
Other examples：
```
make lint HELP=y
make integrationtest HELP=y
make crossbuild HELP=y
make smallbuild HELP=y
make e2e HELP=y
make clean HELP=y
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,111 +1,233 @@
...
+	@echo "$$TEST_HELP_INFO"
+else
+test: 
+	hack/make-rules/test.sh $(WHAT)
+endif
 
-.PHONY: keadm_lint
-keadm_lint:
-	make -C keadm lint
+LINTS=cloud\
```
Comment (kevin-wangzefeng) [inline Makefile]:
```suggestion
LINTS=cloud \
```

File: Makefile [hunk 2]
```diff
@@ -1,111 +1,233 @@
...
+integrationtest: 
+	hack/make-rules/build.sh edgecore 
+	edge/test/integration/scripts/execute.sh
+endif
 
-.PHONY: edge_cross_build_v8
-edge_cross_build_v8:
-	$(MAKE) -C edge armv8
+CROSSBUILD_COMPONENTS=edgecore\
+						edgesite
```
Comment (kevin-wangzefeng) [inline Makefile]:
```suggestion
	edgesite
```

**Comment by kevin-wangzefeng** [comment]
> set bash to build ckubeedge omponeent

commit message typo

**Comment by kadisi** [comment]
@kevin-wangzefeng done

**Comment by kadisi** [comment]
@fisherxu @chendave PTAL

File: hack/lib/golang.sh [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+  for bin in ${binaries[@]}; do
+    echo "cross buildding $bin GOARM${goarm}"
+    local name="${bin##*/}"
+    if [ "${goarm}" == "8" ]; then
+      set -x
+      GOARCH=arm64 GOOS="linux" CGO_ENABLED=1 CC=aarch64-linux-gnu-gcc go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
+      set +x
+    elif [ "${goarm}" == "7" ]; then
+      set -x
+      GOARCH=arm GOOS="linux" GOARM=6 CGO_ENABLED=1 CC=arm-linux-gnueabi-gcc go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
```
Comment (chendave) [inline hack/lib/golang.sh]: should be GOARM=7 here. GOARM=${goarm}

Comment (kadisi) [reply hack/lib/golang.sh]: yes ,you are right , i will push a new pr.

thanks.

**Pull request merged**